modules/dnn/src/net_impl.hpp

```diff
@@ -59,23 +63,52 @@ struct Net::Impl : public detail::NetImplBase
     bool netWasAllocated;
     bool netWasQuantized;
     bool fusion;
-    bool isAsync;
+    bool isAsync;  // FIXIT: drop
     std::vector<int64> layersTimings;
 
 
-    bool empty() const;
-    void setPreferableBackend(int backendId);
-    void setPreferableTarget(int targetId);
+    virtual bool empty() const;
+    virtual void setPreferableBackend(Net& net, int backendId);
+    virtual void setPreferableTarget(int targetId);
 
     // FIXIT use inheritance
-    Ptr<BackendWrapper> wrap(Mat& host);
+    virtual Ptr<BackendWrapper> wrap(Mat& host);
+
+
+    virtual void clear();
 
 
-    void clear();
+    virtual void validateBackendAndTarget();
 
     void setUpNet(const std::vector<LayerPin>& blobsToKeep_ = std::vector<LayerPin>());
 
 
+    virtual Ptr<Layer> createLayerInstance(const LayerData& ld) const
```

Comment:
why is input parameter marked as const?

Reply:
It is `const reference`. I don't see a problem here, as we don't plan to modify input's content.

Reply:
Below const_cast is used to get rid of that, that's what bugged me.

Reply:
There is design problem of underlying DNN API (modifying inputs). Some of them were fixed here: #21437 , but not all of them.